root/static/scripts/edit/URLCleanup.js

```diff
@@ -1205,7 +1205,10 @@ const CLEANUPS = {
     match: [new RegExp('^(https?://)?([^/]+\\.)?itunes\\.apple\\.com/', 'i')],
     type: LINK_TYPES.downloadpurchase,
     clean: function (url) {
-      return url.replace(/^https?:\/\/(?:geo\.)?itunes\.apple\.com\/([a-z]{2}\/)?(artist|album|audiobook|music-video|podcast|preorder)\/(?:[^?#\/]+\/)?(?:id)?([0-9]+)(?:\?.*)?$/, 'https://itunes.apple.com/$1$2/id$3');
+      url = url.replace(/^https?:\/\/(?:geo\.)?itunes\.apple\.com\/([a-z]{2}\/)?(artist|album|audiobook|author|music-video|podcast|preorder)\/(?:[^?#\/]+\/)?(?:id)?([0-9]+)(?:\?.*)?$/, 'https://itunes.apple.com/$1$2/id$3');
+      // Author seems to be a different interface for artist with the same ID
+      url = url.replace(/itunes\.apple\.com\/([a-z]{2}\/)?author/, 'itunes.apple.com/$1artist');
```

Comment:
This defeats cleanup for every other URL part: scheme, host, `id` prefix, query, fragment.
* The first `replace` statement should not have `author` removed.
* The second `replace`’s pattern should preferably affect clean URLs only and can be slightly simplified by capturing a larger group, as follows:
```suggestion
      url = url.replace(/^(https:\/\/itunes\.apple\.com(?:\/[a-z]{2})?)\/author\//, '$1/artist/');
```

Reply:
Wait, isn't that what it does? The first is the same as before, but allowing author, and the second takes the clean url and just switches author for artist. I guess it might even be safe to literally do "if author then say artist" since there shouldn't be any other "author" in the URL pattern, but.

Reply:
like with `http://screwed.itunes.apple.com/ko/author/dj-author/7`? :clown_face:

Reply:
But aren't we already dropping that earlier when doing https://itunes.apple.com/$1$2/id$3 ?

Reply:
No, because of `screwed`.

Reply:
Huh, I see now, but why do we even accept those urls in the first place? :/

Reply:
It is denied by the function `validate` which is called after the function `clean`.